contracts/0.8.9/oracle/ValidatorsExitBus.sol

```diff
@@ -0,0 +1,785 @@
+// SPDX-FileCopyrightText: 2025 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.9;
+
+import {AccessControlEnumerable} from "../utils/access/AccessControlEnumerable.sol";
+import {UnstructuredStorage} from "../lib/UnstructuredStorage.sol";
+import {ILidoLocator} from "../../common/interfaces/ILidoLocator.sol";
+import {Versioned} from "../utils/Versioned.sol";
+import {ExitRequestLimitData, ExitLimitUtilsStorage, ExitLimitUtils} from "../lib/ExitLimitUtils.sol";
+import {PausableUntil} from "../utils/PausableUntil.sol";
+import {IValidatorsExitBus} from "../interfaces/IValidatorsExitBus.sol";
+
+interface ITriggerableWithdrawalsGateway {
+    struct ValidatorData {
+        uint256 stakingModuleId;
+        uint256 nodeOperatorId;
+        bytes pubkey;
+    }
+
+    function triggerFullWithdrawals(
+        ValidatorData[] calldata triggerableExitData,
+        address refundRecipient,
+        uint8 exitType
+    ) external payable;
+}
+
+/**
+ * @title ValidatorsExitBus
+ * @notice An on-chain contract that serves as the central infrastructure for managing validator exit requests.
+ * It stores report hashes, emits exit events, and maintains data and tools that enables anyone to prove a validator was requested to exit.
+ * Unlike VEBO, it supports exit reports from a wide range of entities.
+ */
+contract ValidatorsExitBus is IValidatorsExitBus, AccessControlEnumerable, PausableUntil, Versioned {
```

Comment:
contract should be marked as `abstract` because it is not a standalone contract, but only a base contract for `ValidatorsExitBusOracle`

but it is better to completely move it and merge with the `ValidatorsExitBusOracle`

Reply:
I don't think we should merge them. This code is properly separated and has a different context.
Also, I don't want to mark this contract as abstract, because it can be deployed separately without requiring support for the Oracle mechanism.